indexer/packages/postgres/src/stores/subaccount-usernames-table.ts

```diff
@@ -0,0 +1,94 @@
+import { QueryBuilder } from 'objection';
+
+import { DEFAULT_POSTGRES_OPTIONS } from '../constants';
+import {
+  verifyAllRequiredFields,
+  setupBaseQuery,
+} from '../helpers/stores-helpers';
+import Transaction from '../helpers/transaction';
+import SubaccountUsernamesModel from '../models/subaccount-usernames-model';
+import {
+  QueryConfig,
+  SubaccountUsernamesFromDatabase,
+  SubaccountUsernamesQueryConfig,
+  SubaccountUsernamesColumns,
+  SubaccountUsernamesCreateObject,
+  Options,
+  Ordering,
+  QueryableField,
+} from '../types';
+
+export async function findAll(
+  {
+    username,
+    subaccountId,
+    limit,
+  }: SubaccountUsernamesQueryConfig,
+  requiredFields: QueryableField[],
+  options: Options = DEFAULT_POSTGRES_OPTIONS,
+): Promise<SubaccountUsernamesFromDatabase[]> {
+  verifyAllRequiredFields(
+    {
+      username,
+      subaccountId,
+      limit,
+    } as QueryConfig,
+    requiredFields,
+  );
+
+  let baseQuery: QueryBuilder<SubaccountUsernamesModel> = setupBaseQuery<SubaccountUsernamesModel>(
+    SubaccountUsernamesModel,
+    options,
+  );
+
+  if (username) {
+    baseQuery = baseQuery.where(SubaccountUsernamesColumns.username, username);
+  }
+
+  if (subaccountId) {
+    baseQuery = baseQuery.where(SubaccountUsernamesColumns.subaccountId, subaccountId);
+  }
+
+  if (options.orderBy !== undefined) {
+    for (const [column, order] of options.orderBy) {
+      baseQuery = baseQuery.orderBy(
+        column,
+        order,
+      );
+    }
+  } else {
+    baseQuery = baseQuery.orderBy(
+      SubaccountUsernamesColumns.username,
+      Ordering.ASC,
+    );
+  }
+
+  if (limit) {
+    baseQuery = baseQuery.limit(limit);
+  }
+
+  return baseQuery.returning('*');
+}
+
+export async function create(
+  subaccountUsernameToCreate: SubaccountUsernamesCreateObject,
+  options: Options = { txId: undefined },
+): Promise<SubaccountUsernamesFromDatabase> {
+  return SubaccountUsernamesModel.query(
+    Transaction.get(options.txId),
+  ).insert({
+    ...subaccountUsernameToCreate,
+  }).returning('*');
+}
+
+export async function findByUsername(
```

Comment:
Is making username unique add an index?

Reply:
seems to be https://stackoverflow.com/a/29655542